packages/sentry/index.tsx

```diff
@@ -0,0 +1,17 @@
+import * as Sentry from '@sentry/nextjs';
+
+export const sentryFetch: typeof fetch = async (input, init) => {
+   const context = {
+      input,
+      credentials: init?.credentials,
+      headers: init?.headers,
+      method: init?.method,
+      parameters: init?.body,
+   };
+   Sentry.configureScope((scope) => scope.setContext('request', context));
+   return fetch(input, init).catch((error) => {
+      console.error(error);
+      console.error(context);
+      throw error;
```

Comment:
Does using Promise.catch and then throwing here cause any difference in error handling around fetch?

Reply:
I don't think so

https://www.typescriptlang.org/play?#code/AoJw9gtglgzgpgOhHAVnAxgFwBQHI4CGMmcIABHAOaW4CUC6Bm6AFttnLWQLwB8ZAbwBQZUWXRgAdjDAAbRKXAg8AMyghi4pq1wAaCrQDcIsZhbgA7hWMBfembiT2yGFz6CToidLmJZYSjwHSTIAI390AGs9MhcjITsGbTYON35hMXEpGXkERTBlXHhvABMtZhYYzlt4oVBIWEQXOQA3ODxCYlIKajokivZOHnTPLJ9c-MK1DUxynX1q0bNLawT7FkdnOFdhj0zvHL8AoI2Q8LAomLia-tZBtL2xA988kCU8Yqkyxgqq+LtDEA

Reply:
Does this produce an "error" for 400s? Cause our API produces 400s on the norm:
https://ifixit.slack.com/archives/C90126AN4/p1659398584649409

Reply:
I wouldn't think so, but I don't know how to test it easily. If that's something we want, then we could add a `.then()` block and check `response.ok`

> The Promise returned from fetch() won't reject on HTTP error status even if the response is an HTTP 404 or 500. Instead, it will resolve normally (with ok status set to false), and it will only reject on network failure or if anything prevented the request from completing.

https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API#differences_from_jquery

Reply:
Great! Let's make sure that doesn't include *cancellation* i.e. some browsers cancel requests when navigating away or when pages are in the background.